### DISCUSSION TITLE
How to load characters the PCs can choose?

### DISCUSSION THREAD

**Discussion by Stoob** [fariapp/fari-app#180] [Q&A] [answered]
Maybe I am not understanding how to create a game for my players and have them join.  

I create a game and load the characters.  I copy game link and send link to players.  

When they join, they do not see a list of characters to choose. 

If they join as a new character, I cannot afterwards 'assign' a character sheet to them.  This is the experience I see as a GM and PC.

Please let me know if I am missing a button or misunderstanding.  Thank you!

GM creates game and loads scene and characters, copies link and shares to PC over email
![gm](https://user-images.githubusercontent.com/1683473/102697526-34917500-41eb-11eb-84e0-96676da53c7d.png)

PC click link and attempts to join, but there is no list of characters to join as
![player](https://user-images.githubusercontent.com/1683473/102697525-34917500-41eb-11eb-9151-5bb3c25117cc.png)

PC can join a new character, but there is no way to 'assign' character sheet to the PC after they join
![pc-join](https://user-images.githubusercontent.com/1683473/102697591-baadbb80-41eb-11eb-9795-15d36a2ab25d.png)

What am I missing?

**Comment by RPDeshaies** [comment, marked as answer]
Hi!

Yeah I think the flow could be enhanced since things are not... super clear.

Basically the first thing a player should do, before even joining your game, is to create a character.

![](https://gyazo.com/ad1790bd3fc57ea030b5882ebf42f06b.gif)

If you wanted, you could also create a template character on your end, export it and send them the file so that they can import it as a new character.

Once that is done, you can start your new online game and send them a link.

Once they have a link, they only have to select their existing character and join your session with it

![](https://gyazo.com/dc5c4c44a9ac09e2caa23749cf05c1e8.gif)

And bam!

---

I think this flow could be simplified by... for example [showing the "new character" button even on the join screen](https://gyazo.com/c6f092e50256ee5612f99bfc3312a539) for example (the link is a proof of concept on my local dev environment)

Thanks for your questions BTW as they are helping me understand how new users use the app and what they are having issues with.

**Reply by Stoob** [reply]
Thank you so much for explaining.  

What I will do for now is email the PC a copy of their character .json file and have them load it before they join the game.  This works.  In the future there may be other workflows such as 1) join first then 2) choose a character from a list made available by the GM.  Or 1) join first 2) GM assigns character sheet to player.  But these may be difficult to do technically I do not know.

I am glad my comments are interpreted as constructive because that is my intent.   Thank you so much for making it. 

I think that Fari is really great because I do not want to play Fate as a map & mini game on Roll20 or Foundry.   Fari is how Fate is intended to be played as we watch the videos of the creators playing Fate.  I understand software development is an iterative process and user feedback is important.  So am I happy to help and test your software with you.

**Reply by RPDeshaies** [reply]
Some changes were pushed this morning to include a new flow where the "pick an existing character" button gives you the ability to create a new character directly.

Tell me what you think !

**Comment by Stoob** [comment]
Yes I see that there is a new and import buttons when a PC joins.  This helps **thank you**.  One thing I noticed is that after you join as new the character name is blank, so it cannot be clicked to build your character.  Whereas the character icon can be clicked on the right.  A placeholder name such as "New Player X" may be helpful.
![dafds](https://user-images.githubusercontent.com/1683473/102807176-09d22880-4373-11eb-9ad4-73605a9884da.png)

**Reply by RPDeshaies** [reply]
Good point !

**Reply by RPDeshaies** [reply]
Should be good now :)
